vspace/src/main/java/edu/asu/diging/vspace/web/ExhibitionModuleController.java

```diff
@@ -1,23 +1,48 @@
 package edu.asu.diging.vspace.web;
 
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
 
+import edu.asu.diging.vspace.core.model.IContentBlock;
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.ISlide;
 import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.impl.ContentBlockManager;
 
 @Controller
 public class ExhibitionModuleController {
 
-    @Autowired
-    private IModuleManager moduleManager;
+	@Autowired
+	private IModuleManager moduleManager;
 
-    @RequestMapping(value = "/exhibit/module/{id}")
-    public String module(@PathVariable("id") String id, Model model) {
-        model.addAttribute("module", moduleManager.getModule(id));
+	@Autowired
+	private ContentBlockManager contentBlockManager;
 
-        return "module";
-    }
+	@RequestMapping(value = "/exhibit/module/{id}")
+	public String module(@PathVariable("id") String id, Model model, RedirectAttributes attributes) {
+		IModule module=moduleManager.getModule(id);
+		model.addAttribute("module", module);
+		try {
+			List<List<IContentBlock>> slideContents=new ArrayList<List<IContentBlock>>();
+			List<ISlide> slides = module.getStartSequence().getSlides();
+			Iterator<ISlide> iter=slides.iterator();
+			while(iter.hasNext()) {
+				ISlide slide=iter.next();
+				slideContents.add(contentBlockManager.getAllContentBlocks(slide.getId()));
+			}
+			model.addAttribute("slideContents", slideContents);
```

Comment:
why do you add the contents of all slides here? You only need to show the contents of the first slide? Also, you might just want to redirect from this method to a controller that shows a slide (in this case the first in the sequence), so you can reuse it for all the following slides.